lib/Agrammon/OutputFormatter/Excel.pm6

```diff
@@ -70,38 +68,45 @@ sub output-as-excel(
                         my $value = flat-value($raw-value // 'UNDEFINED');
                         my $var-print = $model.output-print($module, $output) ~ ',All';
                         if not $prints or $var-print.split(',') ∩ @print-set {
-                            $n++;
-                            my $unit = $model.output-unit($module, $output, $language);
+                            my $order        = $model.output-labels($module, $output)<sort> || 'no order';
+                            my $unit         = $model.output-unit($module, $output, $language);
                             my $output-label = $language ?? $model.output-labels($module, $output){$language} !! $output;
                             my $unit-label   = $language ?? $model.output-units($module, $output){$language}  !! $unit;
-
-                            if $module ne $last-module {
-                                $output-sheet.set($row, $col + 0, $module);
-                                $last-module = $module;
-                            }
-                            $output-sheet.set($row, $col+1, $q-name);
-                            $output-sheet.set($row, $col+2, $output-label // 'Output: ???');
-                            $output-sheet.set($row, $col+3, $value, :number-format('#,###'));
-                            $output-sheet.set($row, $col+4, $unit-label // 'Unit: ???');
-                            $row++;
+                            @records.push( %( :module($q-name), :$output-label, :$value, :$unit-label, :$order ) );
                             if $include-filters {
                                 if $raw-value ~~ Agrammon::Outputs::FilterGroupCollection && $raw-value.has-filters {
-                                    render-filters($output-sheet, $row, $col, $model, $raw-value, $unit, $language, :$all-filters);
+                                    push-filters(@records, $q-name, $model, $raw-value, $unit-label, $language, $order, :$all-filters);
                                 }
                             }
                         }
                     }
                 }
             }
         }
-        NEXT $row++ if $n; # empty line between sections
     }
 
+    my $col = 0;
+    my $last-module = '';
+    for @records.sort(+*.<order>) -> %rec {
```

Comment:
I think the `+` here will blow up on your "no order" string:

```
$ raku -e 'say <a b c>.sort(+*)'
Cannot convert string to number: base-10 number must begin with valid digits or '.' in '⏏a' (indicated by ⏏)
  in block <unit> at -e line 1
```

Reply:
Maybe use 0 or -1 or Inf (depending on if you want the unordered things first or last)

Reply:
It did blow up and I already fixed it :-) ... I am now using the last order value as default.